### ISSUE TITLE
Provide circuit diagram

### ISSUE THREAD

**Issue by matt-downs** [matt-downs/arduino-oled-auto-gauges#2]
Arduino Pin | Connected to
--|--
A0 | Sensor signal
5V | 5V VCC input from 12v USB adapter & Display VCC & Sensor VCC
GND | 5V GND input from 12v USB adapter & Display GND & Sensor GND
A4 (SDA) | Display SDA
A5 (SCL) | Display SCL

**Comment by matt-downs** [comment]
@gurvir44 I don't have the software installed to generate a schematic right now but this should hopefully help you out 😄

**Comment by gurvir44** [comment]
Is there a way i can email you? I need a bit more help :/

**Comment by half-finished** [comment]
Which part are you stuck with? It worked first time when I followed it 1-1

**Comment by gurvir44** [comment]
I don’t have a oil pressure sensor yet, is there a part number for one you bought?

Sent from Mail<https://go.microsoft.com/fwlink/?LinkId=550986> for Windows 10

**Comment by matt-downs** [comment]
In my case, the previous owner of my car had installed Defi brand digital boost and oil pressure sensors and gauges - I was able to hook into the existing 3 wire sensors. They were like these ones here: 
https://www.defi-shop.com/installation/sensor/

I think I used a multimeter initially to work out that the Defi sensors were powered by 5v which meant I could power and read them straight from the Arduino. The 3 wires were 5v VCC, 5v GND, and the sensor output. I looked at the datasheets for a bunch of 5v sensors like [this one](https://www.reveltronics.com/en/shop/60/6/onboard-computers/oil-fuel-pressure-sensor-150psi-10bar-5v-detail) and just made the assumption that the Defi sensor reads from 0 to 140psi and the output would be between 0.5-4.5v. I did the same guesswork for the Defi boost sensor. 

I believe sensors like these could work too, however I haven't tried them: 
https://www.rallysportdirect.com/part/gauge-sensors-wiring/prs-psebos-prosport-electric-boost-sender
https://www.rallysportdirect.com/part/gauge-sensors-wiring/prs-pssmops-prosport-premium-oil-pressure-fuel-pressure-sensor

**Comment by gurvir44** [comment]
Thank you! I uploaded the command to my Arduino today and it worked. Sorry for the questions, its my first time using a Arduino.

**Comment by gurvir44** [comment]
Sorry to bother again, can you run both boost and oil pressure with one ardiuno? Or do you need a new one.

**Comment by matt-downs** [comment]
In terms of hardware: 
Sensor readings won't be an issue, just hook each one up to a different analogue input. Not too sure how to drive 2 seperate i2c displays from the same Arduino though. The Arduino Nanos are pretty cheap so I just opted for a seperate one to drive each display lol

**Comment by half-finished** [comment]
Yeah just google how to run 2 displays, you have to do some soldering and addressing in the code but it’s pretty easy, you should be able to run as many sensors as you have analog or digital I/O’s

**Comment by gurvir44** [comment]
My sensor outputs from 1-5v. Do i just have to un comment "return (m - 102) / 0.0585;" so there is no scale? Or do i have to "calculate" a scale for 1-5 which is just simply 0-1023n
Edit: Got it!

**Comment by tobynambur** [comment]
Hallo 
how to make the sreen in Bar instat PSI
